src/efi.rs

```diff
@@ -546,3 +591,54 @@ fn find_file_recursive<P: AsRef<Path>>(dir: P, target_file: &str) -> Result<Vec<
 
     Ok(result)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_get_product_name() -> Result<()> {
+        let product_name = get_product_name()?;
```

Comment:
We shouldn't read from the host system in a unit test.  The best thing to do here is what you did in `test_parse_boot_entries()` - the unit test is testing parsing, leaving the "load the data" part outside of the test.

That all said, I do think we should be reading from the `os-release` file anyways, and actually for that...hmm, there are [a few crates out there](https://lib.rs/search?q=os-release), not sure of maintenance status of any of them.  Maybe not worth pulling in something external.  We may even have something in our ecosystem.  Ah yep, see https://github.com/coreos/rpm-ostree/blob/fec15f5e24ce2c44f97f602152ac913134bd3173/rust/src/countme.rs#L67